config/crd/resources.yaml

```diff
@@ -1057,6 +1060,7 @@ spec:
                         image.
                       type: string
                   required:
+                  - cpu_architecture
```

Comment:
mmh, good point on this potentially breaking existing environments. I'm not really a fan of having a "default" value for this data as I think it should always be specified. 

The idea of bumping the version is also not great, as it means we would have to take check the version and take care of the upgrade path. 

It sounds like, for simplicity, having a default would be good enough for now. I think we can have a default on the operator side but we may want to make it required on the service side/configs. What do you think?

- Can we have the operator update the agent service config CR and add the missing field if it's not present? (Essentially do a schema migration for the CR).  Not sure whether this is a good idea. The point would be to allow us to make it a required field in the future.

Reply:
For now the service fallbacks to the default architecture (x86_64) in case of a missing field.
How can we remove it from required list in resources.yaml? iiuc, It was generated from agentserviceconfig_types.